Editor/AGS.Editor/Components/BaseComponentWithFolders.cs

```diff
@@ -153,6 +154,17 @@ protected string AddSingleItem(ItemType item)
             string newNodeID = AddTreeNodeForItem(item);
             return newNodeID;
         }
+
+        // Swaps two items in the flat item list only, keeping their folder location unchanged.
+        protected void SwapItemsInFlatList(int index1, int index2)

```

Comment:
Swapping items in a array is a very useful feature. Consider making this more generic so it's reusable for us for any array. Like an extensions method that could be added to AGS.Types/Utilities.cs for example.

```
public static void Swap<T>(this T[] array, int index1, int index2)
{
    if (index1 == index2)
        return;

    T temp = array[index1];
    array[index1] = array[index2];
    array[index2] = temp;
}
```